tool/state-processor/src/processor.rs

```diff
@@ -0,0 +1,316 @@
+// std
+use std::{
+	fs::File,
+	io::{Read, Write},
+	marker::PhantomData,
+	mem,
+	sync::RwLock,
+};
+// darwinia
+use crate::*;
+// crates.io
+use anyhow::Result;
+use fxhash::FxHashMap;
+use once_cell::sync::Lazy;
+use parity_scale_codec::{Decode, Encode};
+use serde::de::DeserializeOwned;
+// hack-ink
+use subspector::ChainSpec;
+
+pub type Map<V> = FxHashMap<String, V>;
+
+pub static NOW: Lazy<RwLock<u32>> = Lazy::new(|| RwLock::new(0));
+
+pub struct Processor<S> {
+	pub solo_state: State<S>,
+	pub para_state: State<()>,
+	pub shell_state: State<()>,
+	pub shell_chain_spec: ChainSpec,
```

Comment:
A question I was confused about for a while: what is the role of the `shell_state` and `shell_chain_spec`?

`processed.json` is enough for us in my view.

Reply:
state is storage.

chain spec contains spec name, spec version, boot nodes, and etc.

Reply:
Thanks!  You get the extra information besides state kv by `build-spec`, then mem replaces the state part. Make sense to me.